_test/check-exercises.sh

```diff
@@ -20,15 +20,25 @@ fi
 # can't benchmark with a stable compiler; to bench, use
 # $ BENCHMARK=1 rustup run nightly _test/check-exercises.sh
 if [ -n "$BENCHMARK" ]; then
-    files=exercises/*/benches
+    target_dir=benches
 else
-    files=exercises/*/tests
+    target_dir=tests
+fi
+
+current_branch="$(git rev-parse --abbrev-ref HEAD)"
+
+if [ "$current_branch" != "master" ]; then
+	files="$(git diff --name-only master | grep "exercises/" | cut -d '/' -f -2)"
+else
+	files=exercises/*
```

Comment:
Same issue as https://github.com/exercism/rust/pull/691#discussion_r227505746: this will only work from the repo root.